src/Cli/Commands/AddTelemetryOptions.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Product;
+using CommandLine;
+using Microsoft.Extensions.Logging;
+using static Cli.Utils;
+
+namespace Cli.Commands
+{
+    /// <summary>
+    /// Telemetry command options
+    /// </summary>
+    [Verb("add-telemetry", isDefault: false, HelpText = "Add telemtry for Data Api Builder Application", Hidden = false)]
+    public class AddTelemetryOptions : Options
+    {
+        public AddTelemetryOptions(bool appInsightsEnabled, string appInsightsConnString, string? config) : base(config)
+        {
+            AppInsightsEnabled = appInsightsEnabled;
+            AppInsightsConnString = appInsightsConnString;
+        }
+
+        [Option("app-insights-enabled", Default = false, HelpText = "Enable/Disable Application Insights")]
+        public bool AppInsightsEnabled { get; }
+
+        [Option("app-insights-conn-string", HelpText = "Connection string for the Application Insights resource for telemetry data", Required = true)]
```

Comment:
If app-insights-conn-string is provided, wouldn't it mean the user wants to enable appinsights - so in that case why would they need to additionally specify enabled = true, isnt simply specifying the conn-string sufficient to infer?

Reply:
i think both of them should be independent, since there could be cases where conn-string is set but enabled is false to maybe use it in hot reload case, or in case where the user would want to keep the connection string info in the config file but not enable it.